scipy/stats/_qmc.py

```diff
@@ -860,18 +968,56 @@ class LatinHypercube(QMCEngine):
            [6.80338249, 3.08795949],
            [2.65448791, 3.83491828]])
 
+    Alternatively, optimized LHS can be constructed with `optimization`.
+    It should give a sample with a lower discrepancy at a higher computational
+    cost:
+    >>> sampler = qmc.LatinHypercube(d=2, optimization="random-CD")
+    >>> sample = sampler.random(n=5)
+    >>> qmc.discrepancy(sample)
+    0.017579341838451112  # random
+
     """
 
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
+        optimization: Optional[Literal["random-CD"]] = None,
         seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         self.centered = centered
 
+        optimization_methods: Dict[str, Callable] = {
+            "random-CD": self._random_optimal,
+        }
+
+        if optimization is None:
+            self.optimization = self._random
+        elif optimization in optimization_methods:
+            self.optimization = optimization_methods[optimization]
+        else:
+            raise ValueError(f"{optimization!r} is not a valid optimization"
```

Comment:
```suggestion
        optimization_methods: Dict[str, Callable] = {
            None: self._random,
            "random-CD": self._random_optimal,
        }

        try:
            self.optimization = optimization_methods[optimization]
        except KeyError:
            raise ValueError(f"{optimization!r} is not a valid optimization"
```

Reply:
Better yes 🙌 I will just add `raise from` to have a better trace back.